public/static/app/perfdiscovery/PerfDiscoveryDataService.js

```diff
@@ -110,6 +113,7 @@ mciModule.factory('PerfDiscoveryDataService', function (
       .omit(_.isString)
       .each(function (speed, threads) {
         var data = {
+          all_metrics: getAllMetrics(speed),
```

Comment:
*all_metrics* only applies to each individual result.

The metric names can vary in the stream. 

~Is that correct / what one would expect?~ From the comment on line 137, this would appear to be the case.

Reply:
The name is a bit confusing, but I think given that it's a property on a single test I think it's unambiguous enough. We end up concatenating the list of metrics for the entire list that the user is able to choose from

Reply:
Yeah, there are / have been issues where you the known metrics vary depending on what task result is being processed.

It's good to know the lists are eventually merged.